src/main/java/frc/robot/gamepieces/GamePieceController.java

```diff
@@ -88,30 +140,143 @@ void processGamePieceState(
       LOGGER.debug("Backward Camera");
       camera.backward();
     }
+    updateGamePieces();
+  }
+
+  public void updateGamePieces() {
+    // Update all systems
+    if (RobotMap.HAS_SHOOTER)
+      shooterSM.step();
 
-    switch (mode) {
+    if (RobotMap.HAS_INDEXER)
+      indexerSM.step();
 
-      case DEFENSE:
-        break;
+    // roller controls
+    if (RobotMap.HAS_INTAKE) {
+      if (armPosition) {
+        intake.setIntakeArm(IntakerArm.ARM_UP);
+      } else {
+        intake.setIntakeArm(IntakerArm.ARM_DOWN);
+      }
+
+      if (rollerStateIN) {
+        intake.setIntakeRoller(IntakerRollers.ROLLERS_IN);
+      } else if (rollerStateOUT) {
+        intake.setIntakeRoller(IntakerRollers.ROLLERS_OUT);
+      } else {
+        intake.setIntakeRoller(IntakerRollers.ROLLERS_OFF);
+      }
 
-      case CARGO:
-      default:
-        LOGGER.error("Should always have a game piece mode.");
     }
+  }
 
-    updateGamePieces();
+  public enum DriverInput {
+    FORCE_TRUE, FORCE_FALSE, FORCE_AUTO_TRUE, FORCE_AUTO_FALSE, USE_DRIVER_INPUT
   }
 
-  void updateGamePieces() {
-    // Update all systems
+  DriverInput forceCellsForward = DriverInput.USE_DRIVER_INPUT;
+  DriverInput forceCellsReverse = DriverInput.USE_DRIVER_INPUT;
+  DriverInput forceToAuto = DriverInput.USE_DRIVER_INPUT;
+
+  public void cellsForward(DriverInput mode) {
```

Comment:
These functions should all be renamed to the pattern "setCellsForward", not "cellsForward".